xla/stream_executor/cuda/cuda_blas_lt.cc

```diff
@@ -131,33 +138,31 @@ tsl::Status BlasLt::Init() {
 }
 
 /*static*/ tsl::StatusOr<BlasLt::MatrixLayout> BlasLt::MatrixLayout::Create(
-    blas::DataType type, size_t num_rows, size_t num_cols,
-    BlasLt::MatrixLayout::Order order, size_t batch_size,
-    std::optional<int64_t> leading_dim_stride,
-    std::optional<int64_t> batch_stride) {
-  if (!leading_dim_stride) {
-    leading_dim_stride = (order == Order::kRowMajor) ? num_cols : num_rows;
-  }
+      const gpu::MatrixLayout& m) {
 
+  TF_ASSIGN_OR_RETURN(auto type, gpu::AsBlasDataType(m.dtype));      
+
+  // if (!leading_dim_stride) {
```

Comment:
I think it would be better to not leave this code in comments. Either delete, or uncomment. You could add a TODO comment instead if this needs a followup change. But usually just commenting out code looks like something left over from debugging and can confuse the reader of the code.

Reply:
yes, I will remove it: this is a letfover after I changed the prototype of the function BlasLt::MatrixLayout::Create(). Previously, loading_dim_stride and batch_stride were std::optional<>, now they are always set as members of gpu::MatrixLayout datatype, hence this "intelligence" is no longer needed..